### PR TITLE
Removing unused apis phase 2

### PR DESCRIPTION
**Pull request by UnityAlex**

### CODE REVIEW

**Review by joncham — APPROVED** [review]
Are these APIs no longer needed, or migrated to the new managed embedding layer? Mostly asking if these tests have been migrated over to a different place, or if we are fine just deleting them. cc @bholmes @mrvoorhe

**Comment by UnityAlex** [comment]
> Are these APIs no longer needed, or migrated to the new managed embedding layer? Mostly asking if these tests have been migrated over to a different place, or if we are fine just deleting them. cc @bholmes @mrvoorhe

The ones I've removed were already migrated so I figured while I was in there I'd remove them. I didn't remove all the ones that were migrated though.

File: unity/unity-embed-host/CoreCLRHost.cs [hunk 1]
```diff
@@ -708,14 +690,6 @@ public static IntPtr unity_method_get_function_pointer([NativeCallbackType("Mono
         return tClass.MakeArrayType((int)rank).TypeHandleIntPtr();
     }
 
-
-    [return: NativeCallbackType("gint64")]
-    public static long gc_get_heap_size()
```
Comment (mrvoorhe) [inline unity/unity-embed-host/CoreCLRHost.cs:713]: Are you going to remove this from https://github.cds.internal.unity3d.com/unity/neutron/blob/main/External/ScriptingCore/Unity.Private.Scripting/Interop/GCInterop.cs ?  Same question for `gc_max_generation`

Comment (UnityAlex) [reply unity/unity-embed-host/CoreCLRHost.cs:713]: No that one will stay. This change isn't probably completely necessary as I think you are removing all of the embed host stuff anyway right?

Comment (mrvoorhe) [reply unity/unity-embed-host/CoreCLRHost.cs:713]: Yes.

**Pull request merged**